File: Tyche.Android/bet/src/main/java/com/felipearpa/bet/domain/PoolGamblerBetRepository.kt
package com.felipearpa.bet.domain

import com.felipearpa.core.paging.CursorPage

interface PoolGamblerBetRepository {

    suspend fun bet(bet: Bet): Result<PoolGamblerBet>

    suspend fun getPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String? = null,
        searchText: String? = null
    ): Result<CursorPage<PoolGamblerBet>>
}

File: Tyche.iOS/Bet/Sources/Bet/Domain/PoolGamblerBetRepository.swift
import Core

public protocol PoolGamblerBetRepository {
    func getPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerBet>, Error>
    
    func bet(bet: Bet) async -> Result<PoolGamblerBet, Error>
}
